cmake/public/LoadHIP.cmake

```diff
@@ -233,6 +248,23 @@ if(HIP_FOUND)
   set(CMAKE_HCC_FLAGS_RELEASE ${CMAKE_CXX_FLAGS_RELEASE})
   ### Remove setting of Flags when FindHIP.CMake PR #558 is accepted.###
 
+  if(ROCM_VERSION_DEV VERSION_GREATER_EQUAL "6.0.0")
+    set(HIP_PATH ${ROCM_PATH})
+    set(ROCRAND_PATH ${ROCM_PATH})
+    set(HIPRAND_PATH ${ROCM_PATH})
+    set(ROCBLAS_PATH ${ROCM_PATH})
+    set(MIOPEN_PATH ${ROCM_PATH})
+    set(ROCFFT_PATH ${ROCM_PATH})
+    set(HIPFFT_PATH ${ROCM_PATH})
+    set(HIPSPARSE_PATH ${ROCM_PATH})
+    set(RCCL_PATH ${ROCM_PATH})
+    set(ROCPRIM_PATH ${ROCM_PATH})
+    set(HIPCUB_PATH ${ROCM_PATH})
+    set(ROCTHRUST_PATH ${ROCM_PATH})
+    set(HIPSOLVER_PATH ${ROCM_PATH})
+    set(ROCTRACER_PATH ${ROCM_PATH})
+  endif()
```

Comment:
All of these could move into the earlier sections where each of these _PATH variables are defined. But do we still need the earlier sections that allow user to override using ENV variable? @jeffdaily thoughts?

Reply:
With this change here, it overrides anything the user could have set via env var.  I'm not aware of anyone actually setting the env vars to change individual components, but this code was probably a result of copy/paste/edit and less so any careful design or planning.  We probably don't need the above env var stuff.

I do think it would be a good idea to audit the cmake files related to our HIP/ROCm integration, but perhaps not in this branch.  I think the goal here is to just get release/1.13 workable with ROCm 6.0 changes.

Reply:
Totally agree on this PR not being the place to do this cleanup, I was just using this review as a way to identify future cleanup tasks when we file an upstream PR. Okay, so we will get rid of the verbose env var stuff at the beginning of LoadHIP.cmake when we do the cleanup.